### PR TITLE
fix: Add compiler to PolkaVM for hardhat

### PR DESCRIPTION
**Pull request by bee344**
### Description
* Added a plugin for hardhat in order to compile to PolkaVM instead of EVM, for testing against WAH and Kitchensink.
* Now all the substrate-based chains (Astar, Acala, WAH) are forked with chopsticks.
* Added local execution of Kitchensink to avoid the building process and to integrate it to the init script.
* Updated README.
* For now the CCTP tests are skipped when running against a chain that compiles to PolkaVM, untill we override their compiler.
* Added hardhat configs specific to each chain.
* Remove some `sleep` and replaced them with logging and waiting until the process is ready.
* Moved from `if-else` to `case` in `init.sh` for readability.
* Added binaries for both linux and macOS.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -29,7 +30,10 @@ With `--ethereum` being the default option.
 
 When passing `--endpoint` or `-e`, the second argument must be the `http` endpoint
 of the node you are connecting to. If choosing from a preset chain and specifying
-a test, `--` should be passed for the `<URL>` argument.
+a test, `--` should be passed for the `<URL>` argument. You must also set the
+`USE_REVIVE` env var to either `true` or `false`, in order to enable compilation
```
Comment (IkerAlus) [inline README.md:34]: if a user chooses Westend as network and `false`, what does that mean? will that give any meaningful output?

Comment (bee344) [reply README.md:34]: Got to update that bit. If you choose any of the preset chains, the script sets the value of `USE_REVIVE`, so it shouldn't matter. If on the other hand you are using `-e`, then you should set the value of `USE_REVIVE` depending on the chain you are connecting to.

Comment (bee344) [reply README.md:33]: Could you check again now?

File: init.sh [hunk 2]
```diff
@@ -10,142 +10,225 @@ total_failed=0
 
 LOG_DIR="./test-logs"
 mkdir -p $LOG_DIR
+mkdir -p "./output-logs"
+
+chmod +x ./networks/westend/eth-rpc
+chomd +x ./networks/westend/substrate-node
```
Comment (marshacb) [inline init.sh:16]:
```suggestion
chmod +x ./networks/westend/substrate-node
```

**Review by marshacb — APPROVED** [review]
LGTM

**Pull request merged**